### ISSUE TITLE
[ BUG ] Using 'links' option in an entry breaks sitemap generation

### ISSUE THREAD

**Issue by martinsilha** [ekalinin/sitemap.js#397]
**Describe the bug**
When adding the 'links' option to a sitemap entry that is already working, it is causing sitemap creation to fail validation.  From the 'Options you can pass' example:  

```
links: [
    { lang: 'en', url: 'http://test.com/page-1/' },
    { lang: 'ja', url: 'http://test.com/page-1/ja/' }
  ],
```

**Expected behavior**
I expected the sitemap to add the `<xhtml:link xmlns:xhtml="http://www.w3.org/1999/xhtml" rel="alternate" hreflang="en" href="https://url.example" />` entry to the sitemap, which it does.  The problem is that the entire sheet fails validation after doing so.

**Context:**
 - Library Version 7.1.1

**Additional context**
I am currently generating the sitemap with the 'Serve a sitemap from a server and periodically update it' example from this package.

**Comment by duxor** [comment]
the same for me...

**Comment by daniel-minchev** [comment]
bump

**Comment by migbash** [comment]
same for me

**Comment by migbash** [comment]
@martinsilha can you show me your `sitemap` example ? And how does it fail ? Are you not seeing the proper layout in the browser when navigating to sitemap ?

**Comment by thisfiore** [comment]
same here, if I add links I don't get xml anymore but just a string with link sequences without structure

**Comment by burakDayRoqos** [comment]
same for me

**Comment by KasperiP** [comment]
Thanks for the comment above. This fixed the issue:
```ts
const sitemapXml = (await streamToPromise(smStream)).toString();

// Thanks to this comment:
// https://github.com/ekalinin/sitemap.js/issues/397#issuecomment-1438062056
const sitemapXmlHttps = sitemapXml.replace(
new RegExp('http://', 'g'),
'https://',
);
```

**Comment by huntharo** [comment]
Ok, when we say "fails validation" I think people are simply talking about browsers rendering of XML right?

Questions
===
1. If we're talking about a browser rendering:
   a. Which browser are we talking about?  Chrome?  Something else?
   b. Are we opening the file locally or from a web server?
   c. If from a web server, what is the exact value of the `Content-Type` response header and what is the name of the file (e.g. is it `sitemap.html` or `sitemap.xml`)?
3. If we we're talking about something other than browser rendering
    a. please provide a link to the validator

Note that all the Google docs for sitemaps still show all the `xmlns` links as `http://` in their examples.  I'd hesitate to change these to `https://` to solve a browser rendering issue when it may actually break the sitemaps for crawlers.

https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap

**Comment by derduher** [comment]
right, if you change the xmlns attribute to https it will fail validation by the google crawler (at least as of a couple years ago). xmlns looks like a link but is in fact just a specification token. You can go to that in the browser and it will probably redirect you to https but when this was changed several years ago google would flag it as an invalid sitemap.

And yeah gonna need more info on what "fails validation" is.

**Comment by akhlivniukallane** [comment]
@derduher, I feel like this is not a bug; you are both correct. The only problem is browser XML rendering, but the sitemap itself is entirely correct.
